Glob Syntax Implementation

DarkEmpire362 opened the pull request:
Not currently complete, but adds a glob structure, parser, executor, and an incomplete compiler. 

Resolves  #27

DarkEmpire362 committed:
Add initial headers for globs and parser.

DarkEmpire362 committed:
Updated. Added union and sequential parsing.

DarkEmpire362 committed:
Initial debugging

DarkEmpire362 committed:
Added executor and partially implemented compiler

DarkEmpire362 committed:
Reorganized glob.h

DarkEmpire362 committed:
Fixed makefile

DarkEmpire362 committed:
Parser and Compiler Cleanup Pass

DarkEmpire362 committed:
Added fragment compiler

- Also started looking at state machine optimization in closer detail. May wind up needing a dynamic bitset implementation, though also could be overkill.

DarkEmpire362 committed:
Added linking

DarkEmpire362 committed:
Basic executor functions implemented

DarkEmpire362 committed:
Top level matching and bug fixes

DarkEmpire362 commented:
All core logic is implemented. Parser, Compiler, State Machine, and Executor logic are ready for review. Some less critical features are missing (notably executor flags, and the class that would manage them, as well variable extraction syntax). Optimizations, mainly Bit-Set specializations of the executor and state machine, and code readability improvements are to follow.

DarkEmpire362 committed:
Small changes and documentation

- Small string type corrections
- Some more documentation comments
- More clear todo comments

marron-at-work reviewed (changes requested):
Overall looks good. We need some test assets though -- at least running through the basic scenarios.

DarkEmpire362 committed:
Basic unicode support

DarkEmpire362 committed:
Initial Test Cases, halting until Template Class for Executor is done.

DarkEmpire362 committed:
Add Executor Class

- Fixes failing tests

DarkEmpire362 committed:
Starting wildcard tests

- Bug Fix: Parser can no longer add wildcard to literal after "parseLiteral" rewrite

DarkEmpire362 committed:
Wilcard, Union, and Full-Path Tests

- Added a todo file for tracking the remaining work items.

DarkEmpire362 committed:
Substitution tests

- Substitution bug fixes too!
